webpack.config.js

```diff
@@ -12,55 +13,70 @@ const BUILD_DIR = path.resolve(__dirname, 'src/client/public/app');
 const APP_DIR = path.resolve(__dirname, 'src/client/app');
 
 const config = {
-    entry: APP_DIR + "/index.tsx",
-    output: {
-        filename: "bundle.js",
-        path: BUILD_DIR
-    },
-
     // Enable sourcemaps for debugging webpack's output.
-    devtool: "source-map",
-
+    devtool: 'source-map',
+    entry: {
+        application: APP_DIR + "/index.tsx",
+    },
+    cache: {
+        type: "filesystem"
+    },
     resolve: {
+        fallback: {
+            "buffer": require.resolve('buffer/'),
+            "assert": require.resolve('assert/'),
+            "stream": require.resolve('stream-browserify'),
+            "fs": false
+        },
         // Add '.ts' and '.tsx' as resolvable extensions.
-        extensions: [".ts", ".tsx", ".js", ".json"]
+        extensions: [".css", ".ts", ".tsx", ".js", ".jsx", ".json"]
     },
 
     // Ignore warnings about bundle size
     performance: {
         hints: false
     },
-
     module: {
         rules: [
             // All TypeScript ('.ts' or '.tsx') will be handled by 'awesome-typescript-loader'.
             // Also, for development, JavaScript is handled by 'awesome-typescript-loader' and passed to Babel.
-            { test: /\.[jt]sx?$/, exclude: /node_modules/, loader: "awesome-typescript-loader" },
+            { test: /\.[jt]sx?$/, exclude: /node_modules/, use:[{loader: "awesome-typescript-loader"}] },
             // Any remaining JavaScript ('.js' or '.jsx') will be transpiled by Babel, for production uglification.
-            { test: /\/jsx?$/, exclude: /node_modules/, loader: "babel-loader" },
+            { test: /\/jsx?$/, exclude: /node_modules/, use:[{loader: "babel-loader"}] },
 			// CSS stylesheet loader.
-			{ test: /\.css$/, loader: 'style-loader!css-loader' },
+			{ test: /\.css$/, use: [
+                {loader: "style-loader"},
+                {loader: "css-loader"}
+            ], },
+            {
+                test: /\.m?js/,
```

Comment:
the formatting was done to align with documentation for webpackv5. Without this, webpack v5 would not run, as it would not parse properly. As for the test: /\.m?js/, it had to do with babel: https://stackoverflow.com/questions/65922329/babel-cant-resolve-imports-in-it-its-own-source-code
Hopefully babel will later work? with babel v7, though I am not sure.

I will leave a comment just in case to indicate this, and maybe revisit later.